src/main/java/org/cactoos/list/Immutable.java

```diff
@@ -54,20 +65,51 @@
      */
     private final List<T> list;
 
+    /**
+     * Ctor.
+     * @param items Source array
+     */
+    @SafeVarargs
+    public Immutable(final T... items) {
+        this(new IterableOf<>(items));
+    }
+
     /**
      * Ctor.
      * @param src Source collection
      */
     public Immutable(final Collection<T> src) {
-        this(new ListOf<>(src));
+        this(new IterableOf<>(src.iterator()));
     }
 
     /**
      * Ctor.
      * @param src Source list
      */
     public Immutable(final List<T> src) {
-        this.list = new ListOf<>(src);
+        this(new IterableOf<>(src.iterator()));
+    }
+
+    /**
+     * Ctor.
+     * @param src Source iterable
+     */
+    public Immutable(final Iterable<T> src) {
+        this(() -> {
+            final List<T> copy = new LinkedList<>();
```

Comment:
@iakunin I think it is better to use `ArrayList` here as it better suits for reading and `LinkedList` is better for insert/remove etc. As long as the `Immutable` is for read-only, `ArrayList` would be sufficient

Reply:
Ok, that's not a problem. Fixed!